Skip update check on pull request builds

neilenns opened the pull request:
Fixes #1365

Checks to see if the major version of the build is `0` and if so skips the update check.

```
Info	10/28/2023 12:38:24	Skipping update check since this is an unreleased build.
```

Since I was touching the lines close by anyway I also converted two old-style strings to new style string interpolation.

DocMoebiuz commented on Base/AutoUpdateChecker.cs:
```diff
@@ -15,10 +15,18 @@ public static void CheckForUpdate(bool force = false, bool silent = false)
-            string CurVersion = Assembly.GetExecutingAssembly().GetName().Version.ToString();
-            string CommandToSend = "/check /version " + CurVersion + " /cacheId " + trackingParams;
+            var CurVersion = Assembly.GetExecutingAssembly().GetName().Version;
+            var CommandToSend = $"/check /version {CurVersion} /cacheId {trackingParams}";
```
this line could go under the block `if (CurVersion.Major == 0)` because it is only needed for the other cases.

neilenns replied:
Done!

DocMoebiuz reviewed (changes requested):
Just a minor thing. Very cool!

tigert commented:
works!

of course now if you want to work on the update checker feature, you got to remember this change :-)

neilenns commented:
@DocMoebiuz I went through and cleaned up the remaining three nag messages from Visual Studio in this file. Two fields get set to readonly which is easy. It also pointed out that the `force` parameter to the method never actually gets used, so I removed that which required tweaking the call to the method in MainForm.cs.

DocMoebiuz commented on Base/AutoUpdateChecker.cs:
```diff
@@ -9,16 +9,25 @@ namespace MobiFlight.UpdateChecker
-        public static void CheckForUpdate(bool force = false, bool silent = false)
+        static readonly string mobiFlightInstaller = "MobiFlight-Installer.exe";
+        static readonly int UpdateCheckTimeoutInMs = 5000;
+        public static void CheckForUpdate(bool silent = false)
```
you missed one call to CheckForUpdate in line 610. It should remove true now entirely.

neilenns replied:
Oooooops.

DocMoebiuz reviewed (changes requested):
See comment. THANKS!

DocMoebiuz reviewed (approved):
Thanks!
